### PR TITLE
feat(interface-asset): Add Category property to Asset and Collections

### PR DESCRIPTION
**Pull request by sebtinet**
- When creating an IA via the CDK, we can’t give it a category. This is mandatory information when integrating that IA in the Composer.

### CODE REVIEW

File: libs/core/src/lib/decorators/element.decorator.ts [hunk 1]
```diff
@@ -31,6 +31,11 @@ export interface IElementOptions extends IInjectOptions
      * Description displayed in Composer.
      */
     description?: string;
+
+    /**
+    * Category in Composer's Interface Assets panel.
+    */
+    category?: string;
```
Comment (mikaelkurek) [inline libs/core/src/lib/decorators/element.decorator.ts:38]: In the description of the PR, you say that the category is mandatory. Or in the decorator, it is optional.
So the category is optional or mandatory ? 
If the answer is mandatory, there is also the skeleton of the IA to modify and probably the schematics to ask for a category when creating the IA.

**Review by mikaelkurek — CHANGES_REQUESTED** [review]
Not ok for me, the category is mandatory or optional, we have to choose.

File: libs/core/src/lib/decorators/element.decorator.ts (hunk 1 above)
Comment (sebtinet) [reply libs/core/src/lib/decorators/element.decorator.ts]: Ok changes done. Category is now mandatory and we ask for it when creating the IA.

**Review by mikaelkurek — APPROVED** [review]
Ok, seems good to me

**Review by jviolas — APPROVED** [review]

**Review by nicolas-daures — APPROVED** [review]

**Pull request merged**